spec/controllers/selects_collections_spec.rb

```diff
@@ -7,28 +7,27 @@ class SelectsCollectionsController < ApplicationController
 end
 
 describe SelectsCollectionsController, :type => :controller do
+  let(:user) { FactoryGirl.find_or_create(:jill) }
+  let(:dit) { FactoryGirl.find_or_create(:dit) }
   describe "Select Communities" do
-    before :all do
+    before :each do
```

Comment:
I did some experimenting with this and used let statements for everything
that is set up in the before block.  The result:
```
1) SelectsCollectionsController Select Communities Public Access should
return public communities
     Failure/Error: expect(assigns[:user_communities].map(&:id)).to
match_array [community.id, no_edit_community.id]
       expected collection contained:  ["9z902z939", "cv43nw87f"]
       actual collection contained:    []
       the missing elements were:      ["9z902z939", "cv43nw87f"]
     # ./spec/controllers/selects_collections_spec.rb:49:in `block (4
levels) in <top (required)>'

  2) SelectsCollectionsController Select Communities Public Access should
not return public collections
     Failure/Error: expect(assigns[:user_communities].map(&:id)).not_to
include @collection.id
     NoMethodError:
       undefined method `id' for nil:NilClass
     # ./spec/controllers/selects_collections_spec.rb:53:in `block (4
levels) in <top (required)>'

  3) SelectsCollectionsController Select Communities Regular User Read
Access signed in should return only public and read access (edit access
implies read) communities
     Failure/Error: expect(assigns[:user_communities].map(&:id)).to
match_array [community.id, no_edit_community.id]
       expected collection contained:  ["8049g5125", "rx913p93b"]
       actual collection contained:    ["9z902z939", "cv43nw87f"]
       the missing elements were:      ["8049g5125", "rx913p93b"]
       the extra elements were:        ["9z902z939", "cv43nw87f"]
     # ./spec/controllers/selects_collections_spec.rb:68:in `block (5
levels) in <top (required)>'

  4) SelectsCollectionsController Select Communities Regular User Edit
Access signed in should return only public or editable communities
     Failure/Error: expect(assigns[:user_communities].map(&:id)).to
match_array [community.id]
       expected collection contained:  ["qn59q402s"]
       actual collection contained:    ["9z902z939", "rx913p93b"]
       the missing elements were:      ["qn59q402s"]
       the extra elements were:        ["9z902z939", "rx913p93b"]
     # ./spec/controllers/selects_collections_spec.rb:85:in `block (5
levels) in <top (required)>'
```

I suppose it's lazy but I decided to go back to what worked.

On Thu, Jan 14, 2016 at 4:40 PM, Weiwei Shi <notifications@github.com>
wrote:

> In spec/controllers/selects_collections_spec.rb
> <https://github.com/ualbertalib/HydraNorth/pull/924#discussion_r49802994>:
>
> >    describe "Select Communities" do
> > -    before :all do
> > +    before :each do
>
> why the community/collection need to be set up for each test? Can we reuse
> them for all the tests in this spec?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/pull/924/files#r49802994>.
>